src/lib/icli/index.js

```diff
@@ -0,0 +1,354 @@
+ * @param  {Object} validators - map of parameterKey / validation function
+ * @return void
+ */
+function processCliArgs(cliArgs, validations) {
+  // Initialize an object that will contain the final parameters (cli + prompt)
+  let parameters = cliArgsToParameters(cliArgs);
+  validations = validations || [];
+
+  // We verify that arguments provided in the command are correct
+  // For example, check if the provided api-identifier does really exist
+  // If an argument is a comma separated list, we could also transform it into an Array here
+  const validationResult = validateParameters(parameters, validations);
+  if (validationResult !== true) {
+    console.log('\n  ' + icli.format.ko('error') + ':\n    ' + validationResult.join('\n    ') + '\n');
+    process.exit(1);
+  }
+  return parameters;
+}
+
+/**
+ * Verify and transform cli args into parameters
+ * @param  {Object} cliArgs - arguments and options that have been passed to the cli
+ * @return {Object} - parameter values
+ */
+function cliArgsToParameters(cliArgs) {
+  // Initialize an object that will contain the final parameters (cli + prompt)
+  const parameters = {};
+
+  let options = Array.prototype.pop.call(cliArgs);
+  // Convert cli arguments to parameters
+  _.forEach(options._args, (argsDefinition, i) => {
+    parameters[_.camelCase(argsDefinition.name)] = cliArgs[i];
+  });
+  // Convert cli options to parameters
+  _.forEach(options.options, option => {
+    let key = _.camelCase(option.long);
+    parameters[key] = options[key];
+  });
+  return parameters;
+}
+
+/**
+ * Validate a list of parameter values
+ * Used to validate parameters passed to the command before preparing prompt questions
+ * @param  {Object} parameters - parameters that have been passed to the cli
+ * @param  {Object} validations - validation functions
+ * @return {Array|bool} - true if all validations passed, or else an array of error messages
+ */
+function validateParameters(parameters, validations) {
+  let messages = [];
+  _.forEach(parameters, (value, key) => {
+    if (value && validations[key] ) {
+      const validation = validations[key](value);
+      if (validation !== true) {
+        messages = _.concat(messages, validation);
+      }
+    }
+  });
+  return messages.length ? messages : true;
+}
+
+/**
+ * Transform a list of parameter configurations into a list of inquirer questions
+ * @param  {Array} parameters - a list of parameter configurations
+ * @param  {Object} cmdParameterValues - the parameter values that have been set by the command
+ * @return {Array} - a list of inquirer questions
+ */
+function parametersToQuestions(parameters, cmdParameterValues) {
+  const questions = [];
+  _.forEach(parameters, parameter => {
+    // the question parameter is already an inquirer question
+    const question = parameter.question;
+
+    // But we can extend it with data that comes from the parameter configuration
+    question.type = question.type || parameter.type;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Irregular whitespace not allowed](https://www.codacy.com/app/lager/lager/file/2779185918/issues/source?bid=3360491&fileBranchId=3365065#l320)